src/Compilers/Core/Portable/SourceGeneration/ISyntaxHelper.cs

```diff
@@ -60,18 +58,5 @@ internal abstract class AbstractSyntaxHelper : ISyntaxHelper
         public abstract void AddAliases(CompilationOptions options, ArrayBuilder<(string aliasName, string symbolName)> aliases);
 
         public abstract bool ContainsGlobalAliases(SyntaxNode root);
-
-        public bool ContainsAttributeList(SyntaxNode root)
```

Comment:
yaay, we can no avoid this costly walk.  we originally made this a green walk because the red walk was insanely bad.  but it's still high on CPU (40% costs in some traces).  So this is really nice to avoid even looking at trees without attribuets.  and, if a tree has attributes, we still don't need to walk into parts of hte tree that don't.  For example, method bodies almost never have attributes in them.  But we still had to walk in just in case.  Now we almost never would have to.